templates/calico.go

```diff
@@ -515,4 +515,52 @@ spec:
         - key: aws-ippool
           path: aws-ippool.yaml
           {{end}}
+{{if eq .CloudProvider "gce"}}
+## gce stuff here
+---
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: gce-ippool
+  namespace: kube-system
+data:
+  gce-ippool: |-
+    apiVersion: v1
+    kind: ipPool
+    metadata:
+      cidr: {{.ClusterCIDR}}
+    spec:
+      ipip:
```

Comment:
@galal-hussein The difference is here, I can make this part conditional on cloud provider but thought it would be more clear if we kept it separate. Let me know what you want to do.